telemetry/telemetry_test.go

```diff
@@ -19,165 +19,146 @@ import (
-	c.Assert(jsonParsed.Path("instances.1.instanceType").Data().(string), Equals, "tikv") // mocktikv
-	c.Assert(jsonParsed.ExistsP("hardware"), Equals, true)
+	require.NoError(t, err)
+	require.Equal(t, trackingID, jsonParsed.Path("trackingId").Data().(string))
+	require.True(t, jsonParsed.ExistsP("hostExtra.cpuFlags"))
+	require.True(t, jsonParsed.ExistsP("hostExtra.os"))
+	require.Len(t, jsonParsed.Path("instances").Children(), 2)
+	require.Equal(t, "tidb", jsonParsed.Path("instances.0.instanceType").Data().(string))
+	require.Equal(t, "tikv", jsonParsed.Path("instances.1.instanceType").Data().(string))
+	require.True(t, jsonParsed.ExistsP("hardware"))
 
 	_, err = s.se.Execute(context.Background(), "SET @@global.tidb_enable_telemetry = 0")
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 	r, err = telemetry.PreviewUsageData(s.se, s.etcdCluster.RandClient())
-	c.Assert(err, IsNil)
-	c.Assert(r, Equals, "")
+	require.NoError(t, err)
+	require.Equal(t, "", r)
 
 	_, err = s.se.Execute(context.Background(), "SET @@global.tidb_enable_telemetry = 1")
 	config.GetGlobalConfig().EnableTelemetry = false
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
+
 	r, err = telemetry.PreviewUsageData(s.se, s.etcdCluster.RandClient())
-	c.Assert(err, IsNil)
-	c.Assert(r, Equals, "")
+	require.NoError(t, err)
+	require.Equal(t, "", r)
 }
 
-func (s *testSuite) Test03Report(c *C) {
+func TestReport(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("integration.NewClusterV3 will create file contains a colon which is not allowed on Windows")
+	}
+
+	t.Parallel()
+
+	s := newSuite(t)
+	defer s.close()
+
 	config.GetGlobalConfig().EnableTelemetry = false
 	err := telemetry.ReportUsageData(s.se, s.etcdCluster.RandClient())
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 
 	status, err := telemetry.GetTelemetryStatus(s.etcdCluster.RandClient())
-	c.Assert(err, IsNil)
+	require.NoError(t, err)
 
 	jsonParsed, err := gabs.ParseJSON([]byte(status))
-	c.Assert(err, IsNil)
-
-	c.Assert(jsonParsed.Path("is_error").Data().(bool), Equals, true)
-	c.Assert(jsonParsed.Path("error_msg").Data().(string), Equals, "telemetry is disabled")
-	c.Assert(jsonParsed.Path("is_request_sent").Data().(bool), Equals, false)
+	require.NoError(t, err)
+	require.True(t, jsonParsed.Path("is_error").Data().(bool))
+	require.Equal(t, "telemetry is disabled", jsonParsed.Path("error_msg").Data().(string))
+	require.False(t, jsonParsed.Path("is_request_sent").Data().(bool))
 }
 
-func (s *testSuite) TestCTEPreviewAndReport(c *C) {
-	config.GetGlobalConfig().EnableTelemetry = true
-
-	tk := testkit.NewTestKit(c, s.store)
-	tk.MustExec("use test")
-	tk.MustExec("with cte as (select 1) select * from cte")
-	tk.MustExec("with recursive cte as (select 1) select * from cte")
-	tk.MustExec("with recursive cte(n) as (select 1 union select * from cte where n < 5) select * from cte")
-	tk.MustExec("select 1")
+type testSuite struct {
```

Comment:
I wonder if it is better to move utils like `testSuite` into `main_test.go`, it is common anyway. Not a necessary change request.

Reply:
Make sense. Let me think about it later. For this case, I think it is for this file only and tend not to do.